### ISSUE TITLE
Clarify digest validation wording

### ISSUE THREAD

**Issue by jeremyrickard** [deislabs/cnab-spec#254]
Currently, [101-bundle-json.md](https://github.com/deislabs/cnab-spec/blob/master/101-bundle-json.md) says:

```
The contentDigest field MUST contain a digest, in OCI format, to be used to compute the integrity of the image. The calculation of how the image matches the contentDigest is dependent upon image type. (OCI, for example, uses a Merkle tree while VM images are checksums.) If this field is omitted, a runtime is not obligated to validate the image. During bundle development, it may be ideal to omit the contentDigest field and/or skip validation. Once a bundle is ready to be transmitted as a thick or thin bundle, it must have a contentDigest field.
```

At the last CNAB meeting, a question was asked if this wording was clear enough. Does stating "If this field is omitted, a runtime is not obligated to validate the image." clearly mean that if the field is provided, a runtime is obligated to validate the image or should we add wording to the spec to explicitly state this? OR should https://github.com/deislabs/cnab-spec/blob/master/103-bundle-runtime.md be expanded to discuss these requirements?

**Comment by ryanmoran** [comment]
I do not think that the wording is clear. We should say:
1. a digest is not required
2. if a digest is present, a runtime is required to validate

**Comment by jeremyrickard** [comment]
This has implications on answering https://github.com/deislabs/cnab-go/issues/101

**Comment by squillace** [comment]
A digest is currently required when

> Once a bundle is ready to be transmitted as a thick or thin bundle, it must have a contentDigest field

Per the cnab-go comment above, I refer to the explanation that @technosophos laid down in https://github.com/deislabs/cnab-spec/pull/230#issuecomment-514312953:

> "Our intention was to not generate the hash until the packaging phase of the cycle. Generating a schema validation failure here felt awkward, since we'd basically be encouraging developers to add fake hashes during dev in order to get schema validation to pass.

> The CNAB runtime, though, is still required to verify the presence of this field before trusting an image."

So, here "ready for transmission" means "the packaging phase of the cycle". In https://github.com/deislabs/cnab-go/issues/101#issuecomment-524081866 I argued the following should be adopted:

1. a digest is not required _until it is ready for transmission and/or the packaging phase of the cycle_
2. IFF not present, a runtime must fail and notify (as in the `--force` work we discussed earlier) but a runtime MAY accept and explicit override for developers to choose and not be bothered until they reach the "transmission" state. 
3. IFF it is populated, a runtime MUST validate.

I don't like my wording here, but rather I'm trying to be precise about the state that enables a good default secure experience, but in the dev environment the warning/checking can be turned off until they're ready to "package". Did I reach a good place?
